common/flatpak-dir.c

```diff
@@ -5425,6 +5446,9 @@ flatpak_dir_deploy_install (FlatpakDir   *self,
   if (!flatpak_dir_mark_changed (self, error))
     goto out;
 
+  if (!log_deploy_install_event (self, ref, origin, cancellable, &local_error))
+    g_warning ("Failed to log deploy-install event: %s", local_error->message);
```

Comment:
If you’re deliberately not propagating the error here and are returning a warning instead, you should `g_clear_error (&local_error)`.

Reply:
`local_error` will already be cleared by the use of `g_autoptr` above. Or is the proper convention here to always explicitly clear the error?

Reply:
(Ah, I see by the comment below that is is the convention)

Reply:
It’s convention to clear the error here because it’s served its purpose; what we want to avoid is passing the same `local_error` to another call before clearing it, since that would result in setting a `GError` on top of an already-set one.